dvc/tree.py

```diff
@@ -0,0 +1,145 @@
+import errno
+import os
+
+from dvc.utils.compat import open, StringIO
+
+
+class WorkingTree(object):
```

Comment:
I would split it into two files. It should improve readability. Also, consider introducing an abstract class with an interface. It's easier to navigate, see connection between, catch errors with pylint. Also, I think we can move these files/classes into some reasonable place - one can got to scm/git, another should go to fs/utils or even inside fs.